Source/Core/VideoBackends/OGL/ProgramShaderCache.cpp

```diff
@@ -885,6 +885,24 @@ void ProgramShaderCache::CreateHeader()
     }
   }
 
+  std::string framebuffer_fetch_string;
```

Comment:
Nitpick, but use an immediately-evaluated lambda to avoid reassignment:

```
std::string framebuffer_fetch_string = [&g_ogl_config]() {
    switch(g_ogl_config.SupportedFramebufferFetch)
    {
        case ES_FB_FETCH_TYPE::FB_FETCH_EXT:
               return "#extension GL_EXT_shader_framebuffer_fetch: enable\n"
                          "#define FB_FETCH_VALUE real_ocol0\n"
                          "#define FRAGMENT_INOUT inout";
               // etc.
    }
}()
```

Reply:
An interesting suggestion - but I'm currently leaning against it - to my eyes it feels like the non-lambda control flow is more "obvious", unless there's some specific advantage I'm missing?

Reply:
[It's faster](https://gist.github.com/lefticus/21a5b3bcfc3e95e16f89)

Reply:
While true it effectively removes a default string constructor, I feel chasing "small" performance benefits here would be a mistake as it will certainly be massively overshadowed by the cost of the shader generation itself.

I feel optimising for "programmer understanding" instead would be a better focus. Hence my comment on preferring what I consider easier to follow.

Reply:
This is called once on boot. I doubt we need to care about performance here.